src/ergodex-cdk/components/Form/NewForm.tsx

```diff
@@ -0,0 +1,347 @@
+import React, { createContext, FormEvent, ReactNode, useState } from 'react';
+import {
+  BehaviorSubject,
+  combineLatest,
+  map,
+  mapTo,
+  merge,
+  Observable,
+  publishReplay,
+  refCount,
+  Subscription,
+} from 'rxjs';
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type CheckFn = () => {};
+
+export function useForm<T>(param: T): FormGroup<T> {
+  const [form] = useState(new FormGroup<T>(param));
+
+  return form;
+}
+
+interface EventConfig {
+  readonly emitEvent: 'default' | 'system' | 'silent';
```

Comment:
I assume that naming could be like `evetnType`. Now we have a verb that is usually used to define functions.